### ISSUE TITLE
Failed 8.0.2.0 to 8.2.1 upgrade

### ISSUE THREAD

**Issue by mbcladwell** [owncloud/core#21186]
Exception: Updates between multiple major versions and downgrades are unsupported.
How do I upgrade to 8.1?
How can I roll back to 8.0.2?  I can change the 'version' and 'maintenance' entries in the config but no combination allows me to get back to 8.0

**Comment by PVince81** [comment]
How did you upgrade ? Package manager, updater app, manually extract tarball ?

**Comment by mbcladwell** [comment]
Updater app.

**Comment by PVince81** [comment]
@VicDeo does the updater app skip versions ?

You probably had the updater app from 8.0.2 which is quite old and might still skip versions.

**Comment by PVince81** [comment]
The workaround for now is to download the owncloud 8.1.4 tarball and then install it manually: https://doc.owncloud.org/server/8.1/admin_manual/maintenance/manual_upgrade.html#manual-owncloud-upgrade

and then continue from there.

**Comment by mbcladwell** [comment]
OK I will give that a try.  Isn't the error misleading? This is not a major upgrade.
Thanks

**Comment by PVince81** [comment]
It is a major upgrade, since 8.1 the second digit is considered major too, see https://owncloud.org/blog/owncloud-release-model-changes/

**Comment by VicDeo** [comment]
@mbcladwell >Updates between multiple major versions and downgrades are unsupported.

means that your system left untouched.

**Comment by VicDeo** [comment]
@PVince81 
> does the updater app skip versions ?

it is pretty stupid and downloads what is it told to download. See https://github.com/owncloud/appstore-issues/issues/4#issue-39099299

**Comment by craigarno** [comment]
I have a similar issue.  And will try the proposed 8.1.4 solution later as I ran out of time again this evening.

Some time ago I tried to upgrade 7.0.5 to 8.0.2, then discovered 8.0.2 wasn't fully implemented (too many things I use weren't implemented).  7.0.5 still ran, so I continued using 7.0.5.  Now I'm trying to upgrade to 8.2.1 from my current installation.  I backed up MySQL db, config, data per instuctions this evening. I tried 7.0.11 and it didn't like my current database.  Same with 8.2.1.  When I can make more time I'll try 8.1.4.

My installation:
- OpenSUSE 12.2 x64
- Apache/2.2.22 (Linux/SUSE)
- PHP Version 5.5.13
- mysqlnd 5.0.11-dev - 20120503

I have a large number of working services, so "just upgrading" everything surrounding the OS will take several full-time weeks I don't have and accepting down-time which also can't be accepted.  I'd lose things like phone service through Asterisk during an upgrade.

Thank you for this thread.  I am the only user of my OwnCloud 7.0.5 installation, so if I must I can install 8.2.1 fresh into the above environment and re-populate mozilla-sync, Contacts, Calendar, and Files after resetting up the database as a possible solution.  What I find amazing is 7.0.5 still appears to work just fine after a botched 8.0.2 "upgrade".  So I'm stuck with 7.0.5 until I can figure out how to upgrade OwnCloud which has been pretty painful so far starting with the 8.0.2 debacle.

**Comment by RealRancor** [comment]
@craigarno See https://github.com/owncloud/core/issues/20974#issuecomment-162319494 how to do a correct upgrade.

**Comment by craigarno** [comment]
Solved the issue by exporting all contact and calendar data to vCard, iCal, and LDIF formats (vCard saved pictures too).  Then doing a fresh install and database setup for 8.2.2 and reimported all data from vCard and iCal backups.  Had some "timezone" problems with some iCal data, but it seemed to work.  I took this approach because this eliminates the potential problem of database corruption from a prior botched "upgrade".  I had to manually install mozilla-sync and manually update my mozilla-sync user datbase entry, but all is back to working again with 8.2.2.
